Bolts/Common/BFTask.m

```diff
@@ -172,6 +172,75 @@ + (instancetype)taskForCompletionOfAllTasksWithResults:(NSArray<BFTask *> *)task
     }];
 }
 
++ (instancetype)taskForCompletionOfAnyTask:(NSArray<BFTask *> *)tasks
+{
+    __block int32_t total = (int32_t)tasks.count;
+    if (total == 0) {
+        return [self taskWithResult:nil];
+    }
+    
+    __block dispatch_once_t token = 0;
+    __block int32_t cancelled = 0;
+    
+    NSObject *lock = [NSObject new];
+    NSMutableArray *errors = [NSMutableArray new];
+    NSMutableArray *exceptions = [NSMutableArray new];
+    
+    BFTaskCompletionSource *source = [BFTaskCompletionSource taskCompletionSource];
+    for (BFTask *task in tasks) {
+        [task continueWithBlock:^id(BFTask *task) {
+            if (task.exception != nil) {
+                @synchronized(lock) {
+                    [exceptions addObject:task.exception];
+                }
+            }else if (task.error != nil) {
+                @synchronized(lock) {
+                    [errors addObject:task.error];
+                }
+            }else if (task.cancelled) {
+                OSAtomicIncrement32(&cancelled);
+            }
+            
+            if (task.result != nil) {
```

Comment:
Task can finish with a `nil` result, which should still mark the completion of this task to match closer with other methods here.